### ISSUE TITLE
The loss suddenly becomes larger

### ISSUE THREAD

**Issue by csm-coder** [nie-lang/UnsupervisedDeepImageStitching#27]
hello,thanks for your codes. when i train the H_model, always, the loss suddenly becomes so larger. and it then cannot be lowered. can you give me some advice? here are some of my loss Loss curves:
![1](https://user-images.githubusercontent.com/76400924/160272310-ae5f1328-7191-4b1d-a953-fb869680b25f.png)
![2](https://user-images.githubusercontent.com/76400924/160272323-f358a03e-914c-42ac-9c7e-15deaf797255.png)

**Comment by nie-lang** [comment]
Well, I do not meet this problem when I do it in my dataset. (I guess your dataset contains abundant low-overlap image pairs which can make this unuspervised method hard to train)

Maybe you can pre-train the model in a synthetic dataset first, and then fine-tune it on your own dataset.

**Comment by csm-coder** [comment]
Actually i train it on the coco dataset. when i set the batchsize = 2, it doesn't have this problem。but if i change the loss weight, it also have this problem. have you try other loss weight or learn_rate? it seams the model is a little frail。

**Comment by nie-lang** [comment]
In fact, I have tried different weights for the supervised version of this deep homography model. I found the loss weights should meet: w_{loss1} > w_{loss2} >w_{loss3} (see line59 at https://github.com/nie-lang/UnsupervisedDeepImageStitching/blob/main/ImageAlignment/Codes/train_H.py). 

For example, I set w_{loss1} = w_{loss2} =w_{loss3} for the supervised version, but it seems hard to train. Because the lower pyramid should be set larger weight since its result affect the higher pyramid layer directly. For the unsupervised version, the regulation  is the same.

**Comment by csm-coder** [comment]
thanks a lot for your reply。 in fact, i am just trying to set w_{loss1} = w_{loss2} =w_{loss3}, and it truly hard to train as you said.
